File: FloatItPhone/src/main/java/net/geekstools/floatshort/PRO/Automation/RecoveryBluetooth.java
/*
 * Copyright © 2020 By Geeks Empire.
 *
 * Created by Elias Fazel on 3/24/20 1:15 PM
 * Last modified 3/24/20 10:35 AM
 *
 * Licensed Under MIT License.
 * https://opensource.org/licenses/MIT
 */

package net.geekstools.floatshort.PRO.Automation;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;
import android.util.TypedValue;

import androidx.annotation.Nullable;

import net.geekstools.floatshort.PRO.BindServices;
import net.geekstools.floatshort.PRO.Utils.Functions.FunctionsClass;
import net.geekstools.floatshort.PRO.Utils.Functions.PublicVariable;

public class RecoveryBluetooth extends Service {

    @Override
    public int onStartCommand(Intent intent, int flags, final int startId) {
        FunctionsClass functionsClass = new FunctionsClass(getApplicationContext());
        PublicVariable.size = functionsClass.readDefaultPreference("floatingSize", 39);
        PublicVariable.HW = (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, PublicVariable.size, this.getResources().getDisplayMetrics());

        if (getFileStreamPath(".auto" + getClass().getSimpleName().replace("Recovery", "")).exists()
                && getFileStreamPath(".auto" + getClass().getSimpleName().replace("Recovery", "")).isFile()) {
            try {
                String[] PackageNames = functionsClass.readFileLine(
                        ".auto" + getClass().getSimpleName().replace("Recovery", ""));
                if (PackageNames.length > 0) {
                    for (String PackageName : PackageNames) {
                        functionsClass.runUnlimitedBluetooth(PackageName);
                    }

                    Intent steady = new Intent(getApplicationContext(), BindServices.class);
                    steady.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
                    if (functionsClass.returnAPI() < 26) {
                        startService(steady);
                    } else {
                        startForegroundService(steady);
                    }
                }
            } catch (Exception e) {
                e.printStackTrace();
            }
        }
        if (getFileStreamPath(".auto" + getClass().getSimpleName().replace("Recovery", "") + "Category").exists()
                && getFileStreamPath(".auto" + getClass().getSimpleName().replace("Recovery", "") + "Category").isFile()) {
            try {
                String[] CategoryNames = functionsClass.readFileLine(
                        ".auto" + getClass().getSimpleName().replace("Recovery", "") + "Category");
                if (CategoryNames.length > 0) {
                    for (String CategoryName : CategoryNames) {
                        functionsClass.runUnlimitedFolderBluetooth(CategoryName, functionsClass.readFileLine(CategoryName));
                    }

                    Intent steady = new Intent(getApplicationContext(), BindServices.class);
                    steady.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
                    if (functionsClass.returnAPI() < 26) {
                        startService(steady);
                    } else {
                        startForegroundService(steady);
                    }
                }
            } catch (Exception e) {
                e.printStackTrace();
            }
        }
        stopSelf();
        return functionsClass.serviceMode();
    }

    @Override
    public void onCreate() {
        super.onCreate();
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
    }

    @Nullable
    @Override
    public IBinder onBind(Intent intent) {
        return null;
    }
}

File: FloatItPhone/src/main/java/net/geekstools/floatshort/PRO/Automation/RecoveryServices/RecoveryBluetooth.kt
/*
 * Copyright © 2020 By Geeks Empire.
 *
 * Created by Elias Fazel
 * Last modified 5/28/20 8:34 PM
 *
 * Licensed Under MIT License.
 * https://opensource.org/licenses/MIT
 */

package net.geekstools.floatshort.PRO.Automation.RecoveryServices

import android.app.Service
import android.content.Intent
import android.os.IBinder
import android.util.TypedValue
import net.geekstools.floatshort.PRO.Utils.Functions.FunctionsClass
import net.geekstools.floatshort.PRO.Utils.Functions.FunctionsClassIO
import net.geekstools.floatshort.PRO.Utils.Functions.PublicVariable

class RecoveryBluetooth : Service() {

    override fun onBind(intent: Intent?): IBinder? {

        return null
    }

    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
        super.onStartCommand(intent, flags, startId)

        val functionsClass: FunctionsClass = FunctionsClass(applicationContext)
        val functionsClassIO: FunctionsClassIO = FunctionsClassIO(applicationContext)

        PublicVariable.floatingSizeNumber = functionsClass.readDefaultPreference("floatingSize", 39)
        PublicVariable.floatingViewsHW = TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, PublicVariable.floatingSizeNumber.toFloat(), this.resources.displayMetrics).toInt()

        if (getFileStreamPath(".auto" + this@RecoveryBluetooth.javaClass.simpleName.replace("Recovery", "")).exists()
                && getFileStreamPath(".auto" + this@RecoveryBluetooth.javaClass.simpleName.replace("Recovery", "")).isFile) {

            val packageNames = functionsClassIO.readFileLinesAsArray(".auto" + this@RecoveryBluetooth.javaClass.simpleName.replace("Recovery", ""))
            if (!packageNames.isNullOrEmpty()) {
                for (aPackageName in packageNames) {
                    functionsClass.runUnlimitedBluetooth(aPackageName)
                }
            }
        }

        if (getFileStreamPath(".auto" + this@RecoveryBluetooth.javaClass.simpleName.replace("Recovery", "") + "Category").exists()
                && getFileStreamPath(".auto" + this@RecoveryBluetooth.javaClass.simpleName.replace("Recovery", "") + "Category").isFile) {

            val folderNames = functionsClassIO.readFileLinesAsArray(".auto" + this@RecoveryBluetooth.javaClass.simpleName.replace("Recovery", "") + "Category")
            if (!folderNames.isNullOrEmpty()) {
                for (CategoryName in folderNames) {
                    functionsClass.runUnlimitedFolderBluetooth(CategoryName, functionsClassIO.readFileLinesAsArray(CategoryName))
                }
            }
        }

        stopSelf()

        return Service.START_NOT_STICKY
    }
}